### PR TITLE
Begin React Intl

### PR DESCRIPTION
**Pull request by salinawu**
This PR adds react-intl support for localized strings. In the screenshot, you can see the "test japanese string" showing for the /ja site. I also started adding the language picker that I will flesh out more in another PR

Screen Shot 2020-07-03 at 12 04 57 PM

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -93,8 +93,8 @@ exports.createPages = async ({ graphql, actions }) => {
     const curLangPosts = posts.filter(
       (p) => p.node.fields.langKey === post.node.fields.langKey
     )
-    const previous = curLangPosts[index + 1]?.node
-    const next = curLangPosts[index - 1]?.node
+    const previous = curLangPosts[index + 1] && curLangPosts[index + 1].node
```
Comment (salinawu) [inline gatsby-node.js]: for some reason I couldn't build the app locally with the `curLangPosts[index + 1]?.node` notation

Comment (mngyuan) [reply gatsby-node.js]: interesting ... apparently gatsby doesn't yet support optional chaining, so I'm not sure why this builds on my machine. we can just not use this notation for now then

File: src/components/language-provider.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React, { memo } from 'react'
+import { IntlProvider } from 'react-intl'
+
+export const LocalizationProvider = memo(({ children, locale, messages }) => {
```
Comment (mngyuan) [inline src/components/language-provider.js]: how come the name of this file and the component it exports are named differently? should we just use a default export here?

File: src/templates/blog-index.js [hunk 3]
```diff
@@ -1,55 +1,65 @@
...
-                }}
-              />
-            </section>
-          </article>
-        )
-      })}
-    </Layout>
+    <LocalizationProvider
+      locale={getLocale(langKey)}
+      messages={getMessages(langKey)}
```
Comment (mngyuan) [inline src/templates/blog-index.js:23]: the way the messages are loaded from `../strings/index.js` is cool, nj

**Review by mngyuan — APPROVED** [review]
lookin good! sorry it took so long for me to look at this.

File: src/components/language-provider.js (hunk 2 above)
Comment (salinawu) [reply src/components/language-provider.js]: good point - I was originally going to export other things from this file but i'll update it for now to be default, and change the file name too

**Pull request merged**